examples/run_CPMG.py

```diff
@@ -10,116 +10,131 @@
-    2
-    * config_dict["p90_us"]
-    / pi  # evolution during pulse -- see eq 6 of coherence paper
-    + config_dict["deadtime_us"]  # following 90
-    + config_dict["deblank_us"]  # before 180
-)
+prog_p90_us = prog_plen(config_dict["p90_us"])
+prog_p180_us = prog_plen(2 * config_dict["p90_us"])
+# {{{calculate symmetric tau by dividing 2tau by 2
+marker_us = 1.0
+config_dict["tau_us"] = (2*config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms'])/2
+assert (config_dict['tau_us'] > prog_p180_us/pi + marker_us + config_dict['deblank_us'])
+assert (config_dict['deadtime_us'] > config_dict['deblank_us'] + 2*marker_us)
+assert (
+        2 * config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms'] ==
+        2*config_dict['tau_us'])
+print(
+        "If you are measuring on a scope, the time from the start (or end) of one 180 pulse to the next should be %0.1f us"%(2*config_dict['deadtime_us']+1e3*config_dict['echo_acq_ms']+prog_p180_us))
 # }}}
 # {{{check total points
-total_pts = nPoints * nPhaseSteps
-assert total_pts < 2 ** 14, (
-    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384\nyou could try reducing the acq_time_ms to %f"
-    % (total_pts, config_dict["acq_time_ms"] * 16384 / total_pts)
+total_pts = nPoints * nPhaseSteps  # * config_dict['nEchoes']
+assert total_pts < 2**14, (
+    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"
+    % total_pts
 )
 # }}}
-# {{{run cpmg
-# NOTE: Number of segments is nEchoes * nPhaseSteps
-data = run_cpmg(
-    nScans = config_dict["nScans"],
-    indirect_idx = 0,
-    indirect_len = 1,
-    ph1_cyc = ph1_cyc,
-    adcOffset = config_dict["adc_offset"],
-    carrierFreq_MHz = config_dict["carrierFreq_MHz"],
-    nPoints = nPoints,
-    nEchoes = config_dict["nEchoes"],
-    p90_us = config_dict["p90_us"],
-    repetition_us = config_dict["repetition_us"],
-    tau_us = config_dict["tau_us"],
-    SW_kHz = config_dict["SW_kHz"],
-    pad_start_us = pad_start,
-    pad_end_us = pad_end,
-    ret_data = None,
+data = generic(
+    ppg_list=[
+        ("phase_reset", 1),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p90_us, "ph_cyc", ph1_cyc),
+        ("delay", 
+            config_dict["tau_us"]
+            - 2 * prog_p90_us / pi
+            -marker_us
+            -config_dict['deblank_us']),
+        ("marker", "echo_label", config_dict["nEchoes"]),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p180_us, "ph_cyc", ph2_cyc),
+        ("delay", config_dict["deadtime_us"]),
+        ("acquire", config_dict["echo_acq_ms"]),
+        ("delay", 
+            config_dict['deadtime_us']
+            -2* marker_us
+            - config_dict['deblank_us']
+            ),
+        ("jumpto", "echo_label"),
+        ("delay", config_dict["repetition_us"]),
```

Comment:
there were explanatory comments in the code that I provided.  Please include the explanatory comments.

Reply:
included